rust/src/dcerpc/dcerpc_udp.rs

```diff
@@ -294,9 +294,11 @@ pub unsafe extern "C" fn rs_dcerpc_udp_get_tx_cnt(vtx: *mut std::os::raw::c_void
 /// Probe input to see if it looks like DCERPC.
 fn probe(input: &[u8]) -> (bool, bool) {
     match parser::parse_dcerpc_udp_header(input) {
-        Ok((_, hdr)) => {
+        Ok((leftover_bytes, hdr)) => {
             let is_request = hdr.pkt_type == 0x00;
             let is_dcerpc = hdr.rpc_vers == 0x04 &&
+                hdr.fragnum == 0 &&
+                leftover_bytes.len() >= hdr.fraglen as usize &&
```

Comment:
This looks good. This alone seems good enough to detect your test pcap correctly.
In addition to this, following wireshark's detection logic, possible header field to make probing stronger that isn't already covered:
 `pkt_type` must be <= CANCEL_ACK (10) [this update alone fails the detection test on your s-v pcap though]
 
 lmk wdyt?

Reply:
There seem to be more cf https://github.com/secdev/scapy/blob/master/scapy/layers/dcerpc.py#L138 

```
    10: "cancel_ack",
    11: "bind",
    12: "bind_ack",
    13: "bind_nak",
    14: "alter_context",
    15: "alter_context_resp",
    16: "auth3",
    17: "shutdown",
    18: "co_cancel",
    19: "orphaned",
```

Reply:
Indeed there are. That condition check was used by wireshark for the detection so only the first pkt I think (probing). All the other types you mentioned are after a connection is established and some of these are even handled by our dcerpc parser like bind + bindack. This is why I thought of suggesting it.

Reply:
I would go without it in case we probe midstream...